benchmark/framework/containerd_utils.go

```diff
@@ -0,0 +1,304 @@
+		stdoutPipeWriter.Close()
+		stderrPipeReader.Close()
+		stderrPipeWriter.Close()
+		processStatus, err := task.Status(proc.Context)
+		if processStatus.Status != "stopped" {
+			fmt.Printf("Tried to kill task")
+			err = task.Kill(proc.Context, syscall.SIGKILL)
+			if err != nil {
+				fmt.Printf("Error killing task: %v\n", err)
+			}
+		}
+		// stat, err := task.Status(proc.Context)
+		// fmt.Printf("task status before delete %v\n", stat)
+		_, err = task.Delete(proc.Context)
+		if err != nil {
+			fmt.Printf("Error deleting task: %v\n", err)
+		}
+	}
+
+	return &TaskDetails{
+			task,
+			stdoutPipeReader,
+			stderrPipeReader},
+		cleanupFunc,
+		err
+}
+
+func (proc *ContainerdProcess) RunContainerTaskForReadyLine(
+	taskDetails *TaskDetails,
+	readyLine string) (func(), error) {
+	stdoutScanner := bufio.NewScanner(taskDetails.stdoutReader)
+	stderrScanner := bufio.NewScanner(taskDetails.stderrReader)
+
+	exitStatusC, err := taskDetails.task.Wait(proc.Context)
+	if err != nil {
+		return nil, err
+	}
+	resultChannel := make(chan string, 1)
+	go func() {
+		_ = <-exitStatusC
+		resultChannel <- "PROC_EXIT"
+	}()
+	go func() {
+		continueLoop := true
+		for stderrScanner.Scan() && continueLoop {
+			nextLine := stderrScanner.Text()
+			if strings.Contains(nextLine, readyLine) {
+				resultChannel <- "READYLINE_STDERR"
+				continueLoop = false
+			}
+			select {
+			case _ = <-resultChannel:
+				continueLoop = false
+			default:
+				continueLoop = true
+			}
+		}
+	}()
+	go func() {
+		continueLoop := true
+		for stdoutScanner.Scan() && continueLoop {
+			nextLine := stdoutScanner.Text()
+			if strings.Contains(nextLine, readyLine) {
+				resultChannel <- "READYLINE_STDOUT"
+				continueLoop = false
+			}
+			select {
+			case _ = <-resultChannel:
+				continueLoop = false
+			default:
+				continueLoop = true
+			}
+		}
+	}()
+	go func() {
+		timeoutSecs := 180
+		timeoutMaxInterval := timeoutSecs * 10
+		timeoutInterval := 0
+		continueLoop := true
+		for timeoutInterval < timeoutMaxInterval && continueLoop {
+			time.Sleep(100 * time.Millisecond)
+			timeoutInterval += 1
+			select {
+			case _ = <-resultChannel:
+				continueLoop = false
+			default:
+				continueLoop = true
+			}
+		}
+		resultChannel <- "TIMEOUT"
+	}()
+
+	if err := taskDetails.task.Start(proc.Context); err != nil {
+		return nil, err
+	}
+	_ = <-resultChannel
+	cleanupFunc := func() {
+		processStatus, err := taskDetails.task.Status(proc.Context)
+		if processStatus.Status == "running" {
+			err = taskDetails.task.Kill(proc.Context, syscall.SIGKILL)
+			if err != nil {
+				fmt.Printf("Error killing task: %v\n", err)
+			}
+			exitChannel, _ := taskDetails.task.Wait(proc.Context)
+			_ = <-exitChannel
+		}
+	}
+	return cleanupFunc, nil
+}
+
+func GetRemoteOpts(ctx context.Context, platform string) []containerd.RemoteOpt {
+	var opts []containerd.RemoteOpt
+	opts = append(opts, containerd.WithPlatform(platform))
+	opts = append(opts, containerd.WithPullUnpack)
```

Comment:
can replace the two assignments with one `append(opts, containerd.WithPlatform(platform), containerd.WithPullUnpack)`
or just replace the whole thing with
`opts := [2]containerd.RemoteOpt{containerd.WithPlatform(platform), containerd.WithPullUnpack}`